pilot/pkg/config/file/store.go

```diff
@@ -204,7 +204,17 @@ func (s *KubeSource) ApplyContent(name, yamlText string) error {
 		oldSha, found := s.shas[key]
 		if !found || oldSha != r.sha {
 			s.versionCtr++
-			r.config.ResourceVersion = fmt.Sprintf("v%d", s.versionCtr)
+			if !found {
+				r.config.ResourceVersion = fmt.Sprintf("v%d", s.versionCtr)
+			} else {
+				// If the old resource version exists, and we want to overwrite the old object to the new object,
+				// we should set the resource version in the annotation to avoid of conflicting resource reported
```

Comment:
If I understand this correctly, we are allowing subsequent copies of a resource to override the initial copies.  Is that correct?  If so, what determines the order?

Reply:
@therealmitchconnors Yes correct. The newest content will be the one to be analyzed. The logic I refered to was from the old branch before this https://github.com/istio/istio/pull/35764 got merged. Which explained here:

https://github.com/istio/istio/blob/842115039209d938d6bc9f96736f1ca62e2f8f6f/galley/pkg/config/source/kube/inmemory/kubesource.go#L142-L146

And the code is:

https://github.com/istio/istio/blob/842115039209d938d6bc9f96736f1ca62e2f8f6f/galley/pkg/config/source/kube/inmemory/kubesource.go#L160-L166